mongo/options/clientoptions.go

```diff
@@ -865,3 +888,12 @@ func stringSliceContains(source []string, target string) bool {
 	}
 	return false
 }
+
+func extractX509UsernameFromSubject(subject string) string {
```

Comment:
I would comment why we do it this (i.e. reversing the order of pairs because of what MongoDB wants for x509 username vs what Go library provides as subject name)

Reply:
done.